src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/VertexAIEmbeddingClient.java

```diff
@@ -29,24 +31,86 @@ public interface VertexAIEmbeddingService {
     @POST
     @Path("/{modelId}:predict")
     @ClientHeaderParam(name = "Content-Type", value = "application/json")
-    List<float[]> embed(
+    EmbeddingResponse embed(
         @HeaderParam("Authorization") String accessToken,
         @PathParam("modelId") String modelId,
         EmbeddingRequest request);
   }
 
-  private record EmbeddingRequest(List<Content> instances, Options options) {
+  private record EmbeddingRequest(List<Content> instances) {
     public record Content(String content) {}
+  }
+
+  private static class EmbeddingResponse {
+    public EmbeddingResponse() {}
+
+    private List<Prediction> predictions;
+
+    @JsonIgnore private Object metadata;
```

Comment:
Is this to ignore `metadata` from serialization? If so makes sense; although can alternatively put next to `getMetadata()` for same effect (shouldn't matter, devs have different preferences so this is fine :) ).

Reply:
The response from vertexAI has data other than embedding value which are not used. So added JsonIgnore so these data are not deserialized.